### PR TITLE
[JENKINS-61429] Fix drag & drop for existing steps in job & folders forms

### PR DESCRIPTION
**Pull request by fqueiruga**
See [JENKINS-61429](https://issues.jenkins-ci.org/browse/JENKINS-61429).

This fix seems to do the trick. It seems the webpack introduction in 2.217 caused the proxied callback to Behaviour.specify to not run for repeatable items already on the page at load time.

### Test me

    docker run --rm -ti -p 8080:8080 -e ID=4568 jenkins/core-pr-tester

### Proposed changelog entries

* [JENKINS-61429](https://issues.jenkins-ci.org/browse/JENKINS-61429): Fix bug where drag & drop would fail for for previously saved steps in the job configuration form (regression in 2.217).

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@daniel-beck 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [x] There are at least 2 approvals for the pull request and no outstanding requests for change
- [x] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [x] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [x] Proper changelog labels are set so that the changelog can be generated automatically
- [x] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [x] If it would make sense to backport the change to LTS, a JIRA issue should exist and be labeled as `lts-candidate`

### CODE REVIEW

**Review by daniel-beck — APPROVED** [review]
Manually tested in both job config forms (build steps) and multibranch pipeline folders (shared libraries list). Both work fine.

File: war/src/main/js/widgets/config/tabbar.js [hunk 1]
```diff
@@ -5,13 +5,25 @@ import tableMetadata from './model/ConfigTableMetaData';
 import behaviorShim from '../../util/behavior-shim';
 import jenkinsLocalStorage from '../../util/jenkinsLocalStorage';
 
+/**
+ * Extracting this call from outside of the addPageTabs due to a regression
+ * in 2.216/2.217
```
Comment (oleg-nenashev) [inline war/src/main/js/widgets/config/tabbar.js]:
```suggestion
 * in 2.216/2.217 (see JENKINS-61429)
```

**Review by oleg-nenashev — APPROVED** [review]
There is some dark JS magic there, but the explanation looks good. Since @daniel-beck confirms that it actually works, +1 for the pull request. Will need backporting to 2.222.x IMO

**Comment by olivergondza** [comment]
Cannot really comment on the code side, but the fix seems simple enough.

When can we get this released in weekly? 2.222.1 is scheduled for March 25 and the more soaking we get the better.

**Review by olivergondza — APPROVED** [review]

**Comment by daniel-beck** [comment]
I think we should merge into the upcoming weekly over the weekend unless @fqueiruga wants to dig deeper here or we get negative feedback (seriously folks, run this code, don't just admire its beauty!); then an optimistic backport can basically be done on Monday.

**Comment by oleg-nenashev** [comment]
I plan to merge it in 24 hours if no negative feedback

**Review by timja — APPROVED** [review]

**Review by varyvol — APPROVED** [review]
I've been testing the product itself and everywhere drag & drop can be used, it works fine.

**Review by alecharp — APPROVED** [review]

**Review by batmat — APPROVED** [review]
Just verified the fix:
* ran `23ab517cb4` (master) => unable to reorder. For me the behavior is that this scrolls up incontrollably immediately as you start dragging
* then ran `7f19c4de5b` (this PR) => able to reorder. Saving a job, then reopening does preserve the build steps / reopens them as were stored.

**Comment by daniel-beck** [comment]
> For me the behavior is that this scrolls up incontrollably immediately as you start dragging

That's because the "drag placeholder rectangle" is way up in the config form.

**Comment by batmat** [comment]
Just made a pass on `Maintainer checklist` and checked everything as it looked good to me. 

Reminder: we'll merge this in a bit less than 2 hours [as per Oleg's comment yesterday](https://github.com/jenkinsci/jenkins/pull/4568#issuecomment-598136749).

**Comment by bmunozm** [comment]
It is needed a backport to stable-2.222 right?

**Comment by oleg-nenashev** [comment]
@bmunozm yes, I believe so. The Jira ticket has already been marked accordingly

**Pull request merged**